demo/index.html

```diff
@@ -0,0 +1,150 @@
+<!doctype html>
+<html lang="en-US">
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Book Visual Adjustments</title>
+  <style>
+    @media (prefers-color-scheme: dark) {
+      body {
+        color: #ffffff;
+        background: #181818;
+      }
+    }
+    :root {
+      --menuButtonLabelDisplay: block;
+      --menuSliderBg: #151515;
+      --primaryTextColor: #fff;
+      --menuWidth: 320px;
+      --animationTiming: .25s;
+      --iconFillColor: #fff;
+      --iconStrokeColor: #fff;
+      --iconWidth: 28px;
+      --iconHeight: 28px;
+      --activeButtonBg: #282828;
+    }
+
+    html {
+      font: normal 10px "Helvetica Neue", Helvetica, Arial, sans-serif;
+    }
+
+    body {
+      margin: 0;
+    }
+
+    button {
+      height: 2rem;
+      margin: 1rem;
+      grid-column: 1 / 2;
+    }
+
+    main {
+      display: grid;
+      height: 100vh;
+      grid-template-columns: auto 1fr;
+      grid-template-rows: 1fr auto;
+    }
+
+    #menu {
+      position: relative;
+      grid-column: 1 / 2;
+      width: var(--menuWidth);
+    }
+
+    picture {
+      display: block;
+      grid-column: 2 / 3;
+      grid-row: 1 / 3;
+    }
+
+    img {
+      display: block;
+      margin: 0 auto;
+    }
+  </style>
+</head>
+<body>
+  <button>Toggle menu</button>
+  <main>
+    <div id="menu"></div>
+    <picture>
+      <img src="https://ia801302.us.archive.org/BookReader/BookReaderImages.php?zip=/29/items/goody/goody_jp2.zip&file=goody_jp2/goody_0074.jp2&id=goody&scale=4&rotate=0" />
+    </picture>
+  </main>
+
+  <script type="module">
+    import { render } from 'lit-html';
+    import { html, css, LitElement } from 'lit-element';
+    import IAIcons from '@internetarchive/ia-icons';
+    import '@internetarchive/ia-menu-slider/ia-menu-slider';
+    import '../ia-book-visual-adjustments.js';
+
+    customElements.define('ia-icon', class extends IAIcons {
+      static get styles() {
+        return [css`
+          svg {
+            width: var(--iconWidth);
+            height: var(--iconHeight);
+          }
+        `, IAIcons.styles];
+      }
+    });
+
+    const menuSlider = document.createElement('ia-menu-slider');
+    const visualAdjustmentOptions = [{
+      id: 'grayscale',
+      name: 'Grayscale',
+      active: false,
+    }, {
+      id: 'brightness',
+      name: 'Increased brightness',
+      active: false,
+    }, {
+      id: 'contrast',
+      name: 'Increased contrast',
+      active: false,
+    }, {
+      id: 'invert',
+      name: 'Inverted colors (negative image)',
+      active: false,
+    }];
```

Comment:
Oh so is this the configuration and it just shows the `name` with a checkbox that modifies the `active` state? In that you can pass in any arbitrary entry into there and it'll show that option?

Reply:
Yep. Defaults to just toggling state on a checkbox option. The range option is triggered by a `value` property on each option. `min`, `max`, `step` are all optional and have defaults that are asserted in the tests.